### PR TITLE
(#1) Enabler: App Scaffolding

### LINKED ISSUE #1: Enabler: App Scaffolding
# Setup App Scaffolding

Setup the app scaffolding as the glossary has.

> **ESTIMATE** 
Duration: 2-3 days

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* References: *URL to web site with examples*
* Web Page: *Here goes a URL to an existing web page*

### Sub-Tasks
- [x] Create app scaffold from Playground and update as required
- [x] Set initialize parameters
- [x] Wire up api fetch for drug dictionary

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…
* More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,426 @@
-# drug-dictionary-app
-NCI Drug Dictionary Application
+# React App "Scaffold"
```
Comment (arcepaul) [inline README.md:1]: title on the read me should be:
```
# drug-dictionary-app
NCI Drug Dictionary React Application 
```

Comment (seyilonge-nci) [reply README.md]: Updated!

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+// The following lets us run the app in dev not in situ as would normally be the case.
+const appParams = window.APP_PARAMS || {};
+const integrationTestOverrides = window.INT_TEST_APP_PARAMS || {};
+if (process.env.NODE_ENV !== 'production') {
+	//This is DEV
+	const dictSettings = {
+		...appParams,
+		...integrationTestOverrides,
+		searchEndpoint: '/api/sitewidesearch/v1/',
```
Comment (bryanpizzillo) [inline src/index.js:123]: Fix this please.

File: src/views/ErrorBoundary/ErrorPage.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { i18n } from '../../utils';
+
+const ErrorPage = () => {
+	const [{ canonicalHost, language }] = useStateValue();
+	const tracking = useTracking();
+
+	useEffect(() => {
+		const pageTitle = 'Errors Occurred';
+		tracking.trackEvent({
+			event: 'SitewideSearchApp:Load:Error',
```
Comment (bryanpizzillo) [inline src/views/ErrorBoundary/ErrorPage.jsx:14]: Fix this please and make it `DrugDictionaryApp`

File: src/views/ErrorBoundary/PageNotFound.jsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+const PageNotFound = () => {
+	const [{ canonicalHost, language }] = useStateValue();
+	const [searchText, updateSearchText] = useState('');
+	const tracking = useTracking();
+
+	useEffect(() => {
+		const pageTitle = i18n.pageNotFoundTitle[language];
+		tracking.trackEvent({
+			event: 'SitewideSearchApp:Load:PageNotFound',
```
Comment (bryanpizzillo) [inline src/views/ErrorBoundary/PageNotFound.jsx:17]: Fix this please. It should be `DrugDictionaryApp`

File: src/views/ErrorBoundary/__tests__/ErrorPage.test.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+					<ErrorPage />
+				</MockAnalyticsProvider>
+			);
+		});
+
+		const expectedPageTitle = 'An error occurred. Please try again later.';
+		expect(screen.getByText(expectedPageTitle)).toBeInTheDocument();
+	});
+
+	/*test('should show error page title ( Spanish )', async () => {
```
Comment (bryanpizzillo) [inline src/views/ErrorBoundary/__tests__/ErrorPage.test.js:41]: Please remove this

File: src/views/ErrorBoundary/ErrorPage.jsx (hunk 3 above)
Comment (seyilonge-nci) [reply src/views/ErrorBoundary/ErrorPage.jsx]: Arghhh! Done. I'll update the Playground to have this passed in as props.

File: src/views/ErrorBoundary/PageNotFound.jsx (hunk 4 above)
Comment (seyilonge-nci) [reply src/views/ErrorBoundary/PageNotFound.jsx]: Same as above.

File: src/views/ErrorBoundary/__tests__/ErrorPage.test.js (hunk 5 above)
Comment (seyilonge-nci) [reply src/views/ErrorBoundary/__tests__/ErrorPage.test.js]: Done!

File: src/index.js (hunk 2 above)
Comment (seyilonge-nci) [reply src/index.js]: Done!

**Pull request merged**